lib/eslint.js

```diff
@@ -620,9 +620,11 @@ module.exports = (function() {
      *      If this is not set, the filename will default to '<input>' in the rule context.
      * @param {boolean=} saveState Indicates if the state from the last run should be saved.
      *      Mostly useful for testing purposes.
+     * @param {boolean=} disableInlineConfig Disables inline comments' ability to change config once it is set.
+     *      Useful if you want to validate JS without comments overriding rules.
      * @returns {Object[]} The results as an array of messages or null if no messages.
      */
-    api.verify = function(textOrSourceCode, config, filename, saveState) {
+    api.verify = function(textOrSourceCode, config, filename, saveState, disableInlineConfig) {
```

Comment:
Yeah, let's do that separately. What I'd say is left make the third argument (currently `filename`) into the new options object and let it have `filename`, `saveState`, and whatever else.

Then, @tofumatt, we'll ask you to rebase on top of that change for this PR.